### DISCUSSION TITLE
Query multiple data items from the set of policies

### DISCUSSION THREAD

**Discussion by Abhishek-kumar09** [open-policy-agent/community#453] [OPA and Rego] [answered]
I started this discussion yesterday, but after submitting, the github crashed, so if this is coming as duplicate to you, sorry for that.

I'm using opa-golang library to run and evaluate policies on json input.

I want to query multiple objects from the rego-evaluation engine. I tried some approaches, like:

1. 
```
	regoPolicyLoader := rego.Load(policyDir, nil)

	regoEngine, err := rego.New(
		rego.Query("data.hierarchichal_policy;data.network_policy"), // <-----
		regoPolicyLoader,
	).PrepareForEval(ctx)
```
where the query is separated by semicolon, as I seen in some article on internet, but it didn't work with go-library.

2. Providing multiple Queries like this:
```
	regoEngine, err := rego.New(
		rego.Query(data.network_policy),
		rego.Query("data.hierarchichal_policy"),
		regoPolicyLoader,
	).PrepareForEval(ctx)
```
but it resulted into only hierarchical policy result.

### Working ways:
1. evaluate the policy multiple times with different queries and merge the result
2. don't query to opa, just ask for data and unmarshall the response into the struct with the required format before sending it to the user.

In both the working ways, I think there is problem, 1st one has performance problem, and other one is not using the power of opa-queries perfectly.

Is there any other way, I could query multiple data objects after the policy evaluation. My policies are written in the `policy/` directory and I might just require to ask for few items at a time as a result of evaluation.

**Comment by srenatus** [comment, marked as answer]
You could try querying `"x = data.hierarchichal_policy; y = data.network_policy"`, and then check the bindings.
